### PR TITLE
Antagonize the Crucible Agent

### PR DESCRIPTION
**Pull request by jmpesp**
The `agent-antagonist` will stress test the Crucible agent by continuously requesting and deleting regions, randomly taking and deleting snapshots of those regions.

### CODE REVIEW

File: agent-antagonist/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+    match args {
+        Args::Run {
+            agent,
+            dataset,
+            tasks,
+        } => {
+            let log = build_logger();
+
+            let agent = if let Some(agent) = agent {
+                info!(log, "using agent at argument {:?}", agent);
```
Comment (leftwo) [inline agent-antagonist/src/main.rs:92]: Nit: Agent at address?

File: agent-antagonist/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+                )?;
+                let listen_port =
+                    listen_port.split(' ').last().unwrap().to_string();
+                info!(log, "using listen_port {:?}", listen_port);
+
+                format!("[{}]:{}", listen_addr, listen_port).parse()?
+            };
+
+            let dataset = if let Some(dataset) = dataset {
+                info!(log, "using dataset at argument {:?}", dataset);
```
Comment (leftwo) [inline agent-antagonist/src/main.rs:119]: Also Nit, using dataset {:?} maybe?  Maybe for the other one too?

File: agent-antagonist/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+                                }
+
+                                Err(e) => {
+                                    info!(log, "get region {:?} snapshots failed with {:?}", region_id, e);
+                                    break 'outer;
+                                }
+                            }
+
+                            let client = get_client(&agent);
+                            let r = client.region_get_snapshot(&RegionId(region_id.to_string()), &snapshot_id.to_string()).await;
```
Comment (leftwo) [inline agent-antagonist/src/main.rs:218]: Did `cargo fmt` leave this line longer than 80 characters?  Is it longer than 80?

**Review by leftwo — APPROVED** [review]
This is cool, and much needed to it sounds like.

I would like to see this running in CI, but to do that we need a way to loop for some time
or iteration count and that could be a follow on PR along with the buildomat stuff to
run it.

File: agent-antagonist/src/main.rs (hunk 3 above)
Comment (jmpesp) [reply agent-antagonist/src/main.rs:218]: It's definitely longer, I wonder why fmt didn't pick it up?

File: agent-antagonist/src/main.rs (hunk 1 above)
Comment (jmpesp) [reply agent-antagonist/src/main.rs]: removed, see 1137af4

File: agent-antagonist/src/main.rs (hunk 2 above)
Comment (jmpesp) [reply agent-antagonist/src/main.rs]: removed, see 1137af4

**Review by leftwo — COMMENTED** [review]
Did you move the entire loop to a new function just to
reduce the lines to < 80 characters?

**Comment by leftwo** [comment]
> Did you move the entire loop to a new function just to reduce the lines to < 80 characters?

If it's just a move, then go ahead and merge this with main.
But, if there are other changes, let me know and I'll actually look at it. :)

**Comment by jmpesp** [comment]
> Did you move the entire loop to a new function just to reduce the lines to < 80 characters?

I did! Merging now.

**Pull request merged**